home/api_v3.py

```diff
@@ -179,40 +160,77 @@ def listing_view(self, request, *args, **kwargs):
 
         return self.get_paginated_response(serializer.data)
 
-    def get_queryset(self):
-        qa = self.request.query_params.get("qa")
-        queryset = ContentPage.objects.live().prefetch_related("locale")
+    def get_queryset(self) -> Any:
+        live_queryset = ContentPage.objects.live().prefetch_related("locale")
+        combined_queryset = live_queryset
+        draft_queryset = ContentPage.objects.not_live()
 
-        if qa:
-            queryset = queryset | ContentPage.objects.not_live()
-
-        if "web" in self.request.query_params:
-            queryset = queryset.filter(enable_web=True)
-        elif "whatsapp" in self.request.query_params:
-            queryset = queryset.filter(enable_whatsapp=True)
-        elif "sms" in self.request.query_params:
-            queryset = queryset.filter(enable_sms=True)
-        elif "ussd" in self.request.query_params:
-            queryset = queryset.filter(enable_ussd=True)
-        elif "messenger" in self.request.query_params:
-            queryset = queryset.filter(enable_messenger=True)
-        elif "viber" in self.request.query_params:
-            queryset = queryset.filter(enable_viber=True)
+        return_drafts = (
+            self.request.query_params.get("return_drafts", "").lower() == "true"
+        )
+        if return_drafts:
+            draft_queryset = draft_queryset | live_queryset
+            combined_queryset = draft_queryset | live_queryset
```

Comment:
This looks like we're `|`ing `live_queryset` twice.

Reply:
I have reworked this section a bit, please have another look